### ISSUE TITLE
Dungeon : The Arcatraz - Mob: Destroyed Sentinel (21761) (BB #964)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#964]
This issue was migrated from bitbucket.
**Original Reporter:** silvermcd123
**Original Date:** 03.06.2015 19:23:09 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964

<hr>

This mob is invisible and follows players dealing arcane damage with lightning bolts, i suppose the bug is that you can't target it to kill it.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 04.06.2015 21:02:45 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-18599864

<hr>

Fix for it:

```
#!sql

UPDATE `creature_template` SET `unit_flags`='0' WHERE `entry`='21761';
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** desteny
**Original Date:** 05.06.2015 17:39:36 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-18620319

<hr>

im not sure but i think this are sniffed data, but as far as i know those mobs where never there on blizz.

this spawns should be removed.

look at movie ~@2.00 min https://www.youtube.com/watch?v=8lB9vCxUSq4

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** bewtihozzy
**Original Date:** 06.06.2015 14:56:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-18630179

<hr>

Are these mobs suposed to be here? according to [this](https://youtu.be/8lB9vCxUSq4?t=197)

![WoWScrnShot_060615_162936.jpg](https://bitbucket.org/repo/kMeMB/images/3065932120-WoWScrnShot_060615_162936.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.06.2015 17:31:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-18696800

<hr>

Dungeons long overdue for a rewrite. Here we go :)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 10.06.2015 07:47:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-18715975

<hr>

```
#!c++

DELETE FROM `world`.`creature_template` WHERE `entry` = '21761';
INSERT INTO `world`.`creature_template` (`entry`, `heroic_entry`, `KillCredit1`, `KillCredit2`, `modelid_A`, `modelid_A2`, `modelid_H`, `modelid_H2`, `name`, `subname`, `IconName`, `gossip_menu_id`, `minlevel`, `maxlevel`, `minhealth`, `maxhealth`, `minmana`, `maxmana`, `armor`, `faction_A`, `faction_H`, `npcflag`, `speed`, `scale`, `rank`, `mindmg`, `maxdmg`, `dmgschool`, `attackpower`, `baseattacktime`, `rangeattacktime`, `unit_flags`, `dynamicflags`, `family`, `trainer_type`, `trainer_spell`, `class`, `race`, `minrangedmg`, `maxrangedmg`, `rangedattackpower`, `type`, `type_flags`, `lootid`, `pickpocketloot`, `skinloot`, `resistance1`, `resistance2`, `resistance3`, `resistance4`, `resistance5`, `resistance6`, `spell1`, `spell2`, `spell3`, `spell4`, `PetSpellDataId`, `mingold`, `maxgold`, `AIName`, `MovementType`, `InhabitType`, `RacialLeader`, `RegenHealth`, `equipment_id`, `mechanic_immune_mask`, `flags_extra`, `ScriptName`) VALUES ('21761', '0', '0', '0', '19971', '0', '19971', '0', 'Arcatraz Sentinel', NULL, NULL, '0', '70', '70', '10000', '10000', '0', '0', '5000', '16', '16', '0', '1.125', '1', '0', '450', '600', '0', '1500', '2000', '0', '0', '36', '0', '0', '0', '0', '0', '0', '0', '100', '9', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '', '1', '3', '0', '1', '0', '16384', '0', 'npc_destroyed_sentinel');
INSERT INTO `world`.`creature_template` (`entry`, `heroic_entry`, `KillCredit1`, `KillCredit2`, `modelid_A`, `modelid_A2`, `modelid_H`, `modelid_H2`, `name`, `subname`, `IconName`, `gossip_menu_id`, `minlevel`, `maxlevel`, `minhealth`, `maxhealth`, `minmana`, `maxmana`, `armor`, `faction_A`, `faction_H`, `npcflag`, `speed`, `scale`, `rank`, `mindmg`, `maxdmg`, `dmgschool`, `attackpower`, `baseattacktime`, `rangeattacktime`, `unit_flags`, `dynamicflags`, `family`, `trainer_type`, `trainer_spell`, `class`, `race`, `minrangedmg`, `maxrangedmg`, `rangedattackpower`, `type`, `type_flags`, `lootid`, `pickpocketloot`, `skinloot`, `resistance1`, `resistance2`, `resistance3`, `resistance4`, `resistance5`, `resistance6`, `spell1`, `spell2`, `spell3`, `spell4`, `PetSpellDataId`, `mingold`, `maxgold`, `AIName`, `MovementType`, `InhabitType`, `RacialLeader`, `RegenHealth`, `equipment_id`, `mechanic_immune_mask`, `flags_extra`, `ScriptName`) VALUES ('217612', '0', '0', '0', '19971', '0', '19971', '0', 'Arcatraz Sentinel', NULL, NULL, '0', '70', '70', '15000', '15000', '0', '0', '5000', '16', '16', '0', '1.125', '1', '0', '550', '750', '0', '1500', '2000', '0', '0', '36', '0', '0', '0', '0', '0', '0', '0', '100', '9', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '', '1', '3', '0', '1', '0', '16384', '0', ''); 

/*######
## npc_destroyed_sentinel
######*/

struct npc_destroyed_sentinelAI : public ScriptedAI
{
    uint32 LightningTimer;
    uint32 started;

    npc_destroyed_sentinelAI(Creature *c) : ScriptedAI(c) {}

    void Reset()
    {
        me->SetUInt32Value(UNIT_FIELD_BYTES_1, UNIT_STAND_STATE_DEAD);
        me->SetFlag(UNIT_DYNAMIC_FLAGS, 36);
        LightningTimer = 15000;
        me->UpdateEntry(21761);
        started = false;
    }

    void MoveInLineOfSight(Unit* who)
    {
        if (who->GetTypeId() == TYPEID_PLAYER)
        {
            if (me->IsWithinDistInMap(who, 15) && who->isTargetableForAttack() && !started)
            {
                me->SetUInt32Value(UNIT_FIELD_BYTES_1, 0);
                me->RemoveFlag(UNIT_DYNAMIC_FLAGS, 36);
                me->CastSpell(who, 36915, false);
                me->AI()->AttackStart(who);
                started = true;
            }
        }
    }

    void UpdateAI(const uint32 diff)
    {
        if (me->isInCombat() && started)
        {
            if (LightningTimer <= diff)
            {
                me->CastSpell(me->getVictim(), 36915, false);
                LightningTimer = 15000;
            }
            else LightningTimer -= diff;
        }

        DoMeleeAttackIfReady();

        if (!UpdateVictim())
            return;
    }

};
CreatureAI* GetAI_npc_destroyed_sentinel(Creature* pCreature)
{
    return new npc_destroyed_sentinelAI(pCreature);
}

newscript = new Script;
newscript->Name = "npc_destroyed_sentinel";
newscript->GetAI = &GetAI_npc_destroyed_sentinel;
newscript->RegisterSelf();
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 10.06.2015 07:47:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-18715984

<hr>

offlike

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.07.2015 02:45:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/964/#comment-19886185

<hr>

[Core/Scripts] Cleanup of Arcatraz instance scripts.

```
           Fixes #964
```

→ <<cset 689fc809fd46>>
